app/src/main/java/edu/drexel/lapcounter/lapcounter/backend/lapcounter/LapCounterService.java

```diff
@@ -101,4 +106,10 @@ public void reset() {
         initCallbacks();
         mReceiver.attach(this);
     }
+
+    public void updateThreshold(String mac) {
+        DeviceRepository repo = new DeviceRepository(getApplication());
```

Comment:
Is there a reason why you use the DeviceRepository as opposed to the DeviceViewModel?

Reply:
The LapCounterService is a backend module with no UI component, so there is no UI data for the DeviceViewModel to hold.